### PR TITLE
10 be configure oauth flow

### LINKED ISSUE #10: BE - Configure OAuth Flow
Configure OAuth so that we can get tokens from google.

Protect Backend API with Token Authentication

Use Json Web Tokens as bearer token

### PR DESCRIPTION
**Pull request by kfoxIsProgrammer**
Yea this is the OAuth2.0 security of the backend application

image

This is sort of the basic OAuth flow.

The only extra thing is once the the Resource Server responds, create a JWT to return the frontend

### CODE REVIEW

**Comment by kfoxIsProgrammer** [comment]
The build is failing due to the Application.yml not being present. I am looking at how to add environment secrets to the workflow.
The Application.yml holds my API Secret.

**Comment by MillanWang** [comment]
Combine the Exception and Exceptions folders into one

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * Authorized redirect URLS from the frontend
+     */
+    private List<String> authorizedRedirectUris = new ArrayList<>();
+    /**
+     * The Secret for JWT token generation
+     */
+    private String tokenSecret;
+
+    /**
+     * Expirery of the Token in Msec default set in application.yml-> 864000000msec -> 10 days
```
Comment (NosBos) [inline Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/AppConfig.java]: I think its pulled from application.properties, not yaml.

Comment (kfoxIsProgrammer) [reply Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/AppConfig.java]: Resolved

**Review by MillanWang — APPROVED** [review]
Approved based on a 2 hour conference with Kevin, demonstrating how this works

@NosBos  
@Justgo13 
@GolanHassin 
Strongly recommending that all members look over this before getting it all merged.

**Review by NosBos — APPROVED** [review]

**Review by MillanWang — APPROVED** [review]
Double reviewed this over video call. Users integration looks good and the updated testing precedent is nicely set. 

Note to all devs, be sure to add the 
`src/main/resources/application.properties` file to your backend repository from the discord server

**Comment by kfoxIsProgrammer** [comment]
Make sure to add application.properties that is found on the discord
image

**Pull request merged**